Packs/PerimeterX/Integrations/BotDefender/BotDefender.py

```diff
@@ -0,0 +1,361 @@
+
+    else:
+        """
+        Generate an error because the search type is not supported
+        """
+        raise ValueError('Invalid search_type provided')
+
+    readable_output = f'{result}'
+
+    return CommandResults(
+        outputs_prefix='PerimeterX',
+        outputs_key_field='',
+        outputs=result
+    )
+
+
+def ip(client: Client, args: Dict[str, Any], thresholds: Dict[str, Any]):
+    """
+    Collect the details to run an IP Reputation query against the PerimeterX API
+
+    :type ip: ``str``
+    :param ip: Results will be provided for this particular IP
+
+    :return:
+        A ``CommandResults`` object that is then passed to ``return_results``,
+        that contains Domains
+
+    :rtype: ``CommandResults``
+    """
+
+    # Check to make sure we have a query term
+    ip_address = args.get('ip', None)
+    if not ip_address:
+        raise ValueError('No IP Address specified')
+
+    result = client.post_investigate_by_ip(ip_type='true_ip', ip_address=ip_address)
+
+    pXScore = pXScoring()
+
+    pXScore.add_score_to_ip_list(ip_address=ip_address, risk_score=result['maxRiskScore'], thresholds=thresholds)
+
+    readable_output = f'{pXScore.ip_standard_list}'
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='PerimeterX',
+        outputs_key_field='',
+        outputs=result,
+        indicators=pXScore.ip_standard_list
+    )
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    api_key = demisto.params().get('apikey')
+
+    # get the service API url
+    # '/investigate?search=ip:1.1.1.1&tops=path' + api_key
+    base_url = urljoin(demisto.params()['url'], '/v1/bot-defender/')
+    verify_certificate = not demisto.params().get('insecure', False)
+    proxy = demisto.params().get('proxy', False)
+
+    # get the DBot Thresholds
+    thresholds = {
+        "good_threshold": int(demisto.params().get('dbotGoodThreshold')),
+        "suspicious_threshold": int(demisto.params().get('dbotSuspiciousThreshold')),
+        "bad_threshold": int(demisto.params().get('dbotBadThreshold')),
+        "unknown_threshold": 0
+    }
+
+    LOG(f'Command being called is {demisto.command()}')
+    try:
+        headers = {
+            'Authorization': f'Bearer {api_key}',
+            'Content-Type': 'application/json'
+        }
+        client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy)
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            demisto.results(result)
+
+        elif demisto.command() == 'perimeterx-get-investigate-details':
+            return_results(perimeterx_get_investigate_details(client=client, args=demisto.args(), thresholds=thresholds))
+
+        elif demisto.command() == 'ip':
```

Comment:
This command should support a *list* of IPs, please fix it accordingly.

Reply:
Removed command ATM as it will be supported from next version onwards

Reply:
I did not understand, remove this command or not?
If not, this command should support a list of IPs as input.

Reply:
We do not support list of ips